Geometry/HGCalGeometry/src/HGCalGeometry.cc

```diff
@@ -402,7 +402,8 @@ HGCalGeometry::CornersVec HGCalGeometry::getNewCorners(const DetId& detid, bool
       co[i] = GlobalPoint(
           (r + signr[i] * dr) * cos(fi + signf[i] * dfi), (r + signr[i] * dr) * sin(fi + signf[i] * dfi), (v.z() + dz));
     }
-    co[ncorner - 1] = co[0];
+    // Used to pass downstream the thickness of this cell
+    co[ncorner - 1] = GlobalPoint(0, 0, -2 * dz);
```

Comment:
Would you mind to comment on this line? Do you expect something to change from this line? Thx.

Reply:
This is an understanding between geometry and fireworks - in addition to the corners fireworks also needs half the width. Marco has correctly put it in.